src/index.ts

```diff
@@ -0,0 +1,151 @@
+import * as Cypress from 'cypress';
+import RunResult = CypressCommandLine.RunResult;
+import CypressRunResult = CypressCommandLine.CypressRunResult;
+import CypressFailedRunResult = CypressCommandLine.CypressFailedRunResult;
+import BeforeRunDetails = Cypress.BeforeRunDetails;
+import PluginConfigOptions = Cypress.PluginConfigOptions;
+import PluginEvents = Cypress.PluginEvents;
+import Spec = Cypress.Spec;
+import { ConfigOption, JUnitTestSuite } from './type';
+import Reporter, { TestCase, TestSuite } from './reporter';
+
+let reporter: Reporter;
+
+function onBeforeRun(details: BeforeRunDetails) {
+  // The `projectName` is not officially documented in Cypress documentation.
+  // As used in this context, it represents the basename of the current working directory.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  reporter.junitSuite.name = `Cypress Test - ${(details.config as any)?.projectName} ${details.browser?.displayName}`;
```

Comment:
This type assertion to `any` is still bothering me. I had a look at the type definitions and I see there's a way to get the projectName through the global Cypress namespace:

```
Cypress.config('projectName');
// or
Cypress.config().projectName;
```

Would that work? That might be less prone to future breakage?

Reply:
And `details.browser?.displayName` could evaluate to `undefined`. Do we want a sensible fall back value, like just `''`? Or are we ok with the suite being named like `Cypress Test - My Project undefined`?

Reply:
```
Cypress.config('projectName');
// or
Cypress.config().projectName;
```
I tried but got `ReferenceError: Cypress is not defined`. Is `Cypress` accessible in setup stage?

Reply:
Updated to collect `projectName` from `config`.